core/dbt/task/clone.py

```diff
@@ -132,18 +132,12 @@ def before_run(self, adapter, selected_uids: AbstractSet[str]):
 
     @property
     def resource_types(self):
-        if not self.args.resource_types:
-            return REFABLE_NODE_TYPES
-
-        values = set(self.args.resource_types)
-
-        if "all" in values:
-            values.remove("all")
-            values.update(REFABLE_NODE_TYPES)
-
-        values = [NodeType(val) for val in values if val in REFABLE_NODE_TYPES]
+        resource_types = resource_types_from_args(
+            self.args, set(REFABLE_NODE_TYPES), set(REFABLE_NODE_TYPES)
+        )
 
-        return list(values)
+        resource_types = [NodeType(rt) for rt in resource_types if rt in REFABLE_NODE_TYPES]
```

Comment:
perhaps this won't be necessary if `resource_types_from_args` returns a `Set[NodeType]`